Tests/StoreKitUnitTests/PurchasesOrchestratorTests.swift

```diff
@@ -596,6 +596,34 @@ class PurchasesOrchestratorTests: StoreKitConfigTestCase {
         expect(self.backend.invokedPostReceiptDataParameters?.isRestore) == false
     }
 
+    @available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+    func testStoreKit2TransactionListenerDoesNotFinishTransactionIfPostingReceiptFails() async throws {
```

Comment:
we need to also have a test that ensures that it happens _before_ posting. 
Like, even if you successfully post, if you finished the transaction and refresh the receipt and _then_ post successfully to the backend, a consumable is lost forever. 

We **cannot finish a transaction until it after gets successfully posted to the backend**.

Reply:
> we need to also have a test that ensures that it happens before posting.

I assumed that checking that we don't finish the transaction if the receipt post fails was a way to check ordering, because we can't finish until we know that it finished posting.
But I agree an explicit ordering test would be helpful, let me think about how to best do that.

Reply:
@aboedo done, added a test :)
I love being able to create `enum`s inside the test, so the expectation is this:
```swift
expect(operations) == [
    .receiptPost,
    .finishTransaction
]
```

Reply:
awesome! And the syntax is really nice :)
Yeah, to be clear I wasn't questioning the current code, I was just pointing out that without an explicit test for the ordering, we could break it unknowingly in the future.